### ISSUE TITLE
object_name_linter can't detect S3 methods for the `%/%` operator?

### ISSUE THREAD

**Issue by eitsupi** [r-lib/lintr#2851]
``` r
library(lintr)

lint(
  text = r'(
#' @export
`%%.foo` <- function(x, y) TRUE

#' @export
`%/%.foo` <- function(x, y) TRUE

#' @export
`-.foo` <- function(x, y) TRUE
)',
  linters = object_name_linter()
)
#> <text>:6:1: style: [object_name_linter] Variable and function name style should match snake_case or symbols.
#> `%/%.foo` <- function(x, y) TRUE
#> ^~~~~~~~~
```

<sup>Created on 2025-04-23 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

For some reason only `%/%.foo` seems to be complained by the `object_name_linter` in the above example.
